Add support for Python 3.7

Hi there, we've noticed trying to upgrade our Python version to 3.7 that there is no way to install `opencv-python` with it. Would you consider adding it to the build matrice? Would you like a PR for that?

Thanks for the great work 👌

Windows and Linux builds work fine but there's compiler compatibility issue on macOS due to upgrade to OpenCV 3.4.2:

```
/Users/travis/build/[secure]/opencv-python/opencv/3rdparty/protobuf/src/google/protobuf/arena_impl.h:182:10: error: thread-local storage is not supported for the current target

  static GOOGLE_THREAD_LOCAL ThreadCache thread_cache_;
```

Based on googling Xcode 8 should have support for thread-local storage but unfortunately the error persist with Xcode 8 and Xcode 9 environments in Travis. This is something which needs to be solved before I can make a new release.

--- a/.travis.yml
+++ b/.travis.yml
@@ -264,6 +264,12 @@ matrix:
     - os: linux
       env:
         - MB_PYTHON_VERSION=3.7
+        - ENABLE_CONTRIB=0
+        - ENABLE_HEADLESS=0
+        - TEST_DEPENDS=numpy==1.14.5
+    - os: linux
+      env:
+        - MB_PYTHON_VERSION=3.7
         - PLAT=i686
         - ENABLE_CONTRIB=0
         - ENABLE_HEADLESS=0
@@ -329,6 +335,12 @@ matrix:
         - ENABLE_CONTRIB=0
         - ENABLE_HEADLESS=1
         - TEST_DEPENDS=numpy==1.11.3
+    - os: linux
+      env:
+        - MB_PYTHON_VERSION=3.7
+        - ENABLE_CONTRIB=0
+        - ENABLE_HEADLESS=1
+        - TEST_DEPENDS=numpy==1.14.5
     - os: linux
       env:
         - MB_PYTHON_VERSION=3.7
@@ -400,6 +412,12 @@ matrix:
     - os: linux
       env:
         - MB_PYTHON_VERSION=3.7
+        - TEST_DEPENDS=numpy==1.14.5
+        - ENABLE_CONTRIB=1
+        - ENABLE_HEADLESS=0
+    - os: linux
+      env:
+        - MB_PYTHON_VERSION=3.7
         - PLAT=i686
         - TEST_DEPENDS=numpy==1.14.5
         - ENABLE_CONTRIB=1
@@ -464,6 +482,12 @@ matrix:
         - MB_PYTHON_VERSION=3.6
         - PLAT=i686
         - TEST_DEPENDS=numpy==1.11.3
+        - ENABLE_CONTRIB=1
+        - ENABLE_HEADLESS=1
+    - os: linux
+      env:
+        - MB_PYTHON_VERSION=3.7
+        - TEST_DEPENDS=numpy==1.14.5
         - ENABLE_CONTRIB=1
         - ENABLE_HEADLESS=1
     - os: linux

